systemtest/src/main/java/io/strimzi/systemtest/templates/crd/KafkaTemplates.java

```diff
@@ -304,12 +423,70 @@ private static KafkaBuilder defaultKafka(Kafka kafka, String name, int kafkaRepl
                     .endTopicOperator()
                 .endEntityOperator()
                 .endSpec();
+
+            if (withZookeeper) {
+                kafkaBuilder
+                    .editSpec()
+                        .editZookeeper()
+                            // For ZooKeeper using 512Mi is too much and on the other hand 128Mi is causing OOM problem at the start.
+                            .withResources(new ResourceRequirementsBuilder()
+                                .addToLimits("memory", new Quantity("256Mi"))
+                                .addToRequests("memory", new Quantity("256Mi"))
+                                .build())
+                        .endZookeeper()
+                    .endSpec();
+            }
         }
+    }
 
-        return kb;
+    private static void applyDefaultSpecOfKafka(KafkaBuilder kafkaBuilder, int kafkaReplicas) {
+        kafkaBuilder
+            .editSpec()
+                .editKafka()
+                    .withListeners(
+                        new GenericKafkaListenerBuilder()
+                            .withName(TestConstants.PLAIN_LISTENER_DEFAULT_NAME)
+                            .withPort(9092)
+                            .withType(KafkaListenerType.INTERNAL)
+                            .withTls(false)
+                            .build(),
+                        new GenericKafkaListenerBuilder()
+                            .withName(TestConstants.TLS_LISTENER_DEFAULT_NAME)
+                            .withPort(9093)
+                            .withType(KafkaListenerType.INTERNAL)
+                            .withTls(true)
+                            .build())
+                    .withVersion(Environment.ST_KAFKA_VERSION)
+                    .addToConfig("offsets.topic.replication.factor", Math.min(kafkaReplicas, 3))
+                    .addToConfig("transaction.state.log.min.isr", Math.min(kafkaReplicas, 2))
+                    .addToConfig("transaction.state.log.replication.factor", Math.min(kafkaReplicas, 3))
+                    .addToConfig("default.replication.factor", Math.min(kafkaReplicas, 3))
+                    .addToConfig("min.insync.replicas", Math.min(Math.max(kafkaReplicas - 1, 1), 2))
+                .endKafka()
+            .endSpec();
+    }
+
+    private static KafkaBuilder removeFieldsNotRelatedToParticularMode(KafkaBuilder kafkaBuilder, boolean withZookeeper) {
+        Kafka kafka = kafkaBuilder.build();
```

Comment:
is this needed? `.withZookeeper(null)` on builder doesn't work?

Reply:
Hmm I can try that.
I did it this way because IIRC when there was `null`, the default was used.
But it was in case of TopicOperator, so maybe this will be anyway needed for TO.